CHANGES.md

```diff
@@ -5,6 +5,8 @@
 * Effects: improved CPS transform, resulting in lower compilation time and smaller generated code
 * Compiler: separate compilation can now drops unused units when linking (similar to ocamlc). (#1378)
   Feature is disabled by default while dune rules are being fixed. Enable with --enable=auto-link.
+* Compiler: specialize string to js-string conversion for all valid utf8 strings (previously just ascii)
```

Comment:
IIUC I think it would be worthwhile to add here something along the lines:
```
JavaScript files generated by `js_of_ocaml` are now UTF-8 encoded. 
You need to make sure your web server serves them with that encoding or integrate 
them in your HTML with <script src="…" charset="UTF-8"></script>`
```

Reply:
I was considering also adding an option to tune the behavior at compile time.
And I am now wondering whether we should have a runtime check that print an error on the console if the encoding is not the one expected. What do you think ?

Reply:
> I was considering also adding an option to tune the behavior at compile time.

In general configurability is worse for bugreports/maintenance so it's your call :-) I'd rather go for a well advertised change, if not a main version bump.

> And I am now wondering whether we should have a runtime check that print an error on the console if the encoding is not the one expected. What do you think ?

I don't understand how that would/could work. But in any case, IIUC the code has shipped to the client at that point so it's kind of too late.

Reply:
I was hoping that a test / a few test such as the following would work (place toward the top of the file).
```js
if("π" != "\u{3c0}") {
  console.error(..)
}
```

Reply:
Do you thing the generated file should contain a BOM ?

Reply:
> I was hoping that a test / a few test such as the following would work (placed toward the top of the file).
> 
> ```js
> if("π" != "\u{3c0}") {
>   console.error(..)
> }
> ```

That probably only work when the encoding used "include" ASCII.